File: examples/sorting.pas
program BubbleSort;

var
    arr: array[1..10] of integer;
    n, i: integer;

{ Процедура для обмена двух элементов }
procedure swap(var a, b: integer);
var
    temp: integer;
begin
    temp := a;
    a := b;
    b := temp;
end;

{ Процедура сортировки пузырьком }
procedure bubbleSort(var a: array[1..10] of integer; size: integer);
var
    i, j: integer;
    swapped: boolean;
begin
    for i := 1 to size - 1 do
    begin
        swapped := false;
        for j := 1 to size - i do
        begin
            if a[j] > a[j + 1] then
            begin
                swap(a[j], a[j + 1]);
                swapped := true;
            end;
        end;
        
        { Если не было обменов, массив уже отсортирован }
        if not swapped then
            break;
    end
end;

{ Процедура вывода массива }
procedure printArray(a: array[1..10] of integer; size: integer);
var
    i: integer;
begin
    for i := 1 to size do
        write(a[i], ' ');
    writeln;
end;

begin
    write('Введите количество элементов (1-10): ');
    readln(n);
    
    if (n < 1) or (n > 10) then
    begin
        writeln('Ошибка: количество должно быть от 1 до 10');
    end
    else
    begin
        writeln('Введите ', n, ' элементов:');
        for i := 1 to n do
        begin
            write('arr[', i, '] = ');
            readln(arr[i]);
        end;
        
        write('Исходный массив: ');
        printArray(arr, n);
        
        bubbleSort(arr, n);
        
        write('Отсортированный массив: ');
        printArray(arr, n);
    end
end.

File: examples/sorting.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

void swap(int& a, int& b);

void bubbleSort(int a[], int size);

void printArray(int a[], int size);

int main() {
    int arr[10];
    int n;
    int i;

    cout << "Введите количество элементов (1-10): ";
    cin >> n;
    if (((n < 1) || (n > 10))) {
        {
            cout << "Ошибка: количество должно быть от 1 до 10" << endl;
        }
    } else {
        {
            cout << "Введите " << n << " элементов:" << endl;
            for (int i = 1; i <= n; i++) {
                {
                    cout << "arr[" << i << "] = ";
                    cin >> arr[(i - 1)];
                }
            }
            cout << "Исходный массив: ";
            printArray(arr, n);
            bubbleSort(arr, n);
            cout << "Отсортированный массив: ";
            printArray(arr, n);
        }
    }
    return 0;
}

void swap(int& a, int& b) {
    int temp;

    temp = a;
    a = b;
    b = temp;
}

void bubbleSort(int a[], int size) {
    int i;
    int j;
    bool swapped;

    for (int i = 1; i <= (size - 1); i++) {
        {
            swapped = false;
            for (int j = 1; j <= (size - i); j++) {
                {
                    if ((a[(j - 1)] > a[((j + 1) - 1)])) {
                        {
                            swap(a[(j - 1)], a[((j + 1) - 1)]);
                            swapped = true;
                        }
                    }
                }
            }
            if (!(swapped)) {
                break;
            }
        }
    }
}

void printArray(int a[], int size) {
    int i;

    for (int i = 1; i <= size; i++) {
        cout << a[(i - 1)] << ' ';
    }
    cout << endl;
}
